fe/src/main/java/org/apache/doris/external/EsStateStore.java

```diff
@@ -17,94 +17,95 @@
 import org.apache.doris.catalog.SinglePartitionInfo;
 import org.apache.doris.catalog.Table;
 import org.apache.doris.catalog.Table.TableType;
-import org.apache.doris.common.AnalysisException;
 import org.apache.doris.common.Config;
 import org.apache.doris.common.DdlException;
 import org.apache.doris.common.util.MasterDaemon;
-
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.collect.Maps;
-import com.google.common.collect.Range;
-
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.json.JSONObject;
 
-import java.io.IOException;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-
-import okhttp3.Authenticator;
-import okhttp3.Call;
-import okhttp3.Credentials;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.Response;
-import okhttp3.Route;
-
 
 /**
  * it is used to call es api to get shard allocation state
  */
 public class EsStateStore extends MasterDaemon {
+
     private static final Logger LOG = LogManager.getLogger(EsStateStore.class);
 
     private Map<Long, EsTable> esTables;
 
+    private Map<Long, EsRestClient> esClients;
+
     public EsStateStore() {
         super("es state store", Config.es_state_sync_interval_second * 1000);
         esTables = Maps.newConcurrentMap();
+        esClients = Maps.newConcurrentMap();
     }
 
     public void registerTable(EsTable esTable) {
         if (Catalog.isCheckpointThread()) {
             return;
         }
         esTables.put(esTable.getId(), esTable);
-        LOG.info("register a new table [{}] to sync list", esTable.toString());
+        esClients.put(esTable.getId(),
+            new EsRestClient(esTable.getSeeds(), esTable.getUserName(), esTable.getPasswd()));
+        LOG.info("register a new table [{}] to sync list", esTable);
     }
 
     public void deRegisterTable(long tableId) {
         esTables.remove(tableId);
+        esClients.remove(tableId);
         LOG.info("deregister table [{}] from sync list", tableId);
     }
-    
+
     @Override
     protected void runAfterCatalogReady() {
         for (EsTable esTable : esTables.values()) {
             try {
-                EsRestClient client = new EsRestClient(esTable.getSeeds(),
-                        esTable.getUserName(), esTable.getPasswd());
-                // if user not specify the es version, try to get the remote cluster versoin
-                // in the future, we maybe need this version
-                String indexMetaData = client.getIndexMetaData(esTable.getIndexName());
-                if (indexMetaData == null) {
+                EsRestClient client = esClients.get(esTable.getId());
+
+                if (esTable.isKeywordSniffEnable() || esTable.isDocValueScanEnable()) {
+                    JSONObject properties = client.getIndexProperties(esTable.getIndexName(), esTable.getMappingType());
```

Comment:
I think maybe this file should not appear `JSONxxxx`

Reply:
We don't know the full structure of properties, and not using JSONObject feels like no other good way to do it